### PR TITLE
Wait for process to stop before terminating or detaching. Also fixes …

### PR DESCRIPTION
**Pull request by paulmaybee**
…process reference cleanup on abort event.
Issue #520  
@gregg-miskelly @jacdavis @faxue-msft

### CODE REVIEW

File: src/MICore/Debugger.cs [hunk 1]
```diff
@@ -617,10 +620,13 @@ protected bool IsCoreDump
             _detaching = true;
             if (ProcessState == ProcessState.Running)
```
Comment (gregg-miskelly) [inline src/MICore/Debugger.cs]: You also want: `&& this.MICommandFactory.Mode != MIMode.Clrdbg`

File: src/MICore/Debugger.cs [hunk 2]
```diff
@@ -603,10 +603,13 @@ protected bool IsCoreDump
                 _terminating = true;
                 if (ProcessState == ProcessState.Running && this.MICommandFactory.Mode != MIMode.Clrdbg)
                 {
+                    await AddInternalBreakAction(() => MICommandFactory.Terminate());
                     await CmdBreak(BreakRequest.Async);
```
Comment (gregg-miskelly) [inline src/MICore/Debugger.cs]: AddInternalBreakAction already does a CmdBreak, so can't we remove this?

Comment (paulmaybee) [inline src/MICore/Debugger.cs]: yes, done

File: src/MICore/Debugger.cs (hunk 1 above)
Comment (paulmaybee) [inline src/MICore/Debugger.cs]: done

**Comment by gregg-miskelly** [comment]
LGTM. Merge when ready...

**Comment by paulmaybee** [comment]
Currently not planning on going to ask. If early access users start complaining about hitting it all the time we may change our minds.

**Pull request merged**